proto/spire/api/server/local_authority/v1/local_authority.proto

```diff
@@ -0,0 +1,174 @@
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintJWTAuthority(TaintJWTAuthorityRequest) returns (TaintJWTAuthorityResponse);
+
+        // RevokeJWTAuthority revokes the previously active JWT authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        //
+        // If a previously active JWT authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeJWTAuthority(RevokeJWTAuthorityRequest) returns (RevokeJWTAuthorityResponse);
+
+        // GetX509AuthorityState returns the state of all locally configured
+        // X.509 authorities.
+        rpc GetX509AuthorityState(GetX509AuthorityStateRequest) returns (GetX509AuthorityStateResponse);
+
+        // PrepareX509Authority prepares a new X.509 authority for use by
+        // generating a new key and injecting the resulting CA certificate into
+        // the bundle. This action will  propagate the new CA cluster-wide.
+        rpc PrepareX509Authority(PrepareX509AuthorityRequest) returns (PrepareX509AuthorityResponse);
+
+        
+        // ActivateX509Authority activates a prepared X.509 authority for use,
+        // which will cause it to be used for all X.509 signing operations
+        // serviced by this server going forward. If a new X.509 authority has
+        // not already been prepared, a FailedPrecondition error will be returned.
+        rpc ActivateX509Authority(ActivateX509AuthorityRequest) returns (ActivateX509AuthorityResponse);
+
+        // TaintX509Authority marks the previously active X.509 authority as
+        // being tainted. SPIRE Agents observing an authority to be tainted
+        // will perform proactive rotations of any key material related to
+        // the tainted authority. The result of this action will be observed
+        // cluster-wide.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintX509Authority(TaintX509AuthorityRequest) returns (TaintX509AuthorityResponse);
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
```

Comment:
Since we have only two slots per authority type, we can infer which key we want to take the action on based on the type of action. For example, we only ever want to revoke non-active key, and we only ever want to activate a prepared key. In this way, I've managed to sidestep the need to (re)invent key identifiers, however I have still included an opaque slot name for ease of use.

Reply:
That makes sense to me.